src/components/search/searchPageSearchForm/SearchPageSearchForm.tsx

```diff
@@ -71,10 +80,19 @@ function SearchPageSearchForm() {
   const suggestions =
     data?.unifiedSearchCompletionSuggestions?.suggestions ?? [];
 
+  const show = getURLSearchParamsFromAsPath(router.asPath).get("show");
+  const showMode = getShowMode(show);
+
   return (
-    <div className={styles.searchArea}>
-      <Text variant="h1">Mitä etsit?</Text>
-      <form role="search" className={styles.form} onSubmit={handleSubmit}>
+    <div>
+      {showMode === ShowMode.LIST && <Text variant="h1">Mitä etsit?</Text>}
+      <form
+        role="search"
+        className={classNames(styles.form, {
+          [styles.mapMode]: showMode === ShowMode.MAP,
+        })}
+        onSubmit={handleSubmit}
+      >
```

Comment:
I think that this component doesn't need to know whether it is being displayed in a list or in a map. It guides future changes in a direction which may make the component harder to maintain because we would be creating a more opaque separation of concerns. I think that this component's job should be to allow the user to input search terms and then refresh the page with the new criteria after the user communicates that they are done.

If there are problems with margin, then the definition of those margins should perhaps be moved to the containing component. If some text needs to be hidden conditionally then we could consider whether we can do that by using a prop.